torch/nn/modules/module.py

```diff
@@ -586,87 +613,162 @@ def state_dict(self, destination=None, prefix='', keep_vars=False):
         """
         if destination is None:
             destination = OrderedDict()
+            destination._versions = OrderedDict()
+        if hasattr(destination, '_versions'):
+            destination._versions[prefix + 'version'] = self._version
         for name, param in self._parameters.items():
             if param is not None:
                 destination[prefix + name] = param if keep_vars else param.data
         for name, buf in self._buffers.items():
             if buf is not None:
                 destination[prefix + name] = buf
-        for name, module in self._modules.items():
-            if module is not None:
-                module.state_dict(destination, prefix + name + '.', keep_vars=keep_vars)
+        if not no_child:
+            for name, module in self._modules.items():
+                if module is not None:
+                    module.state_dict(destination, prefix + name + '.', keep_vars=keep_vars)
         return destination
 
-    def _load_state_dict_key_mismatch(self, full_name, name, is_missing):
-        r"""This is called in :meth:`~torch.nn.Module.load_state_dict` when
-        there is state dict key mismatch in ``strict=True`` mode. This method
-        can be overridden by subclasses to raise class-specific errors.
-
-        When :attr:`is_missing` is ``True``, :attr:`full_name` can not be found in
-        the dict being loaded. When :attr:`is_missing` is ``False``,
-        :attr:`full_name` is unexpected in the dict being loaded.
+    def load_local_state_dict(self, local_state_dict, version, strict):
```

Comment:
Can we call it `load_from_state_dict`? I'd like to avoid having multiple different concepts called `state_dict`.

Reply:
I'm not sure if we want to add this to the API just yet. E.g. `_version` is a private attribute, but it is necessary if someone would like to use it. On the other hand, making every `nn.Module` have a `.version` attr doesn't seem nice. Let's just keep it private so it can get into the release quickly, and later we can think of ways to clean it up